circuit/src/nullifier.rs

```diff
@@ -71,27 +85,32 @@ impl From<&CircuitInputs> for Nullifier {
 #[derive(Debug, Clone)]
 pub struct NullifierTargets {
     hash: HashOutTarget,
-    preimage: Vec<Target>,
+    secret: Vec<Target>,
+    funding_nonce: Target,
+    funding_account: Vec<Target>,
 }
 
 impl NullifierTargets {
     pub fn new(builder: &mut CircuitBuilder<F, D>) -> Self {
+        // TODO: reuse target from other fragment here
```

Comment:
Just to comment on this, what needs to be done is:
1. Create the target, somewhere in `CircuitTargets`.
2. Modify the `new` function, in this nullifier and the unspendable account, to take in the secret target.